geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -128,8 +128,10 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     while (running.get) {
       queue.take() match {
         case update: CreateOrUpdate =>
-          fireEvent(KafkaFeatureEvent.changed(this, update.feature))
-          featureCache.createOrUpdateFeature(update)
+          if(query.getFilter.evaluate(update.feature)) {
```

Comment:
That's right.  I forget that we only get the DELETE filter back.

From the end-users' point of view, then...
1.  an INSERT/UPDATE will send me an event iff it satisfies my filter;
2.  a DELETE will always sent me an event, even if what was removed does not satisfy my filter

Am I understanding this correctly?